jupyter_server/services/contents/handlers.py

```diff
@@ -116,18 +117,26 @@ async def get(self, path=""):
         content = int(content_str or "")
 
         if not cm.allow_hidden and await ensure_async(cm.is_hidden(path)):
-            raise web.HTTPError(404, f"file or directory {path!r} does not exist")
-
-        model = await ensure_async(
-            self.contents_manager.get(
-                path=path,
-                type=type,
-                format=format,
-                content=content,
+            self.set_status(HTTPStatus.NOT_FOUND)
+            self.write(f"file or directory {path!r} does not exist")
+            await self.finish()
+        try:
+            model = await ensure_async(
+                self.contents_manager.get(
+                    path=path,
+                    type=type,
+                    format=format,
+                    content=content,
+                )
             )
-        )
-        validate_model(model, expect_content=content)
-        self._finish_model(model, location=False)
+            validate_model(model, expect_content=content)
+            self._finish_model(model, location=False)
+        except web.HTTPError as exc:
```

Comment:
I would suggest adding some comments here to explain why a web.HTTPError needs to be caught here, as it doesn't **make any difference to the client**

Mostly in https://github.com/jupyter-server/jupyter_server/issues/1328, we don't want a meaningless stack throw, which means that the situation here is fully under control.